File: java/src/main/java/com/cartenz/core/BaseSubscriber.java
package com.cartenz.core;

import java.util.ArrayList;
import java.util.List;

import rx.Subscription;

public abstract class BaseSubscriber {

    private List<Subscription> subscriptions = new ArrayList<>();

    public void addSubscription(Subscription subscription) {
        subscriptions.add(subscription);
    }

    public void finishSubscriber() {
        for (Subscription sub : subscriptions) {
            sub.unsubscribe();
        }
    }

    public List<Subscription> getSubscriptions() {
        return subscriptions;
    }
}

File: kotlin/src/main/java/com/cartenz/kotlin_core/BaseSubscriber.kt
package com.cartenz.kotlin_core

import rx.Subscription
import java.util.*

abstract class BaseSubscriber {

    companion object {
        private val subscriptions = ArrayList<Subscription>()

        fun addSubscription(subscription: Subscription) {
            subscriptions.add(subscription)
        }

        fun finishSubscriber() {
            for (sub in subscriptions) {
                sub.unsubscribe()
            }
            subscriptions.clear()
        }

        fun getSubscriptions(): List<Subscription> {
            return subscriptions
        }
    }
}